### PR TITLE
Feature/local distances to ground sensor

### PR DESCRIPTION
**Pull request by dtch1997**
Heightfield sensor. 

Senses distance from robot height to ground in a square grid around robot (in robot frame). 
Grid unit length and grid size subject to some tuning.

### CODE REVIEW

File: blind_walking/envs/env_builder.py [hunk 1]
```diff
@@ -55,7 +55,14 @@ def build_regular_env(robot_class,
 
   env_sensor_list = [
     environment_sensors.LastActionSensor(num_actions=a1.NUM_MOTORS),
+<<<<<<< HEAD
```
Comment (jennyzzt) [inline blind_walking/envs/env_builder.py:58]: I think you didn't resolve the mere conflicts

File: blind_walking/envs/sensors/environment_sensors.py [hunk 2]
```diff
@@ -308,3 +308,49 @@ def to_local_frame(dx, dy, yaw):
...
+      env: the environment who invokes this callback function.
+    """
+    self._env = env
+
+  def _get_observation(self) -> _ARRAY:
+    """ Returns the local distances to ground """
+    return self._env.robot.GetLocalDistancesToGround(
+      grid_unit = self.grid_unit, 
+      grid_size = self.grid_size
+    ).reshape(-1)
```
Comment (jennyzzt) [inline blind_walking/envs/sensors/environment_sensors.py:356]: good coding practice to have a newline at the end of the file

File: blind_walking/net/feature_encoder.py [hunk 3]
```diff
@@ -6,6 +6,28 @@
 from stable_baselines3.common.torch_layers import BaseFeaturesExtractor
 from stable_baselines3.common.torch_layers import create_mlp
```
Comment (jennyzzt) [inline blind_walking/net/feature_encoder.py:8]: trivial, but convention is to have two blank lines between the import statements and other code

File: blind_walking/net/feature_encoder.py [hunk 4]
```diff
@@ -6,6 +6,28 @@
...
 
+def _create_depthwise_separable_conv2d_block(in_channels, out_channels):
```
Comment (jennyzzt) [inline blind_walking/net/feature_encoder.py:9]: move to a utilities/utils folder if you want to keep it

File: blind_walking/envs/env_builder.py (hunk 1 above)
Comment (dtch1997) [reply blind_walking/envs/env_builder.py]: thanks for catching that! my bad.

File: blind_walking/envs/locomotion_gym_config.py [hunk 5]
```diff
@@ -45,7 +45,7 @@ class SimulationParameters(object):
   enable_clip_motor_commands = attr.ib(type=bool, default=True)
 
   # 0-plain, 1-heightfield, 2-collapsible_tile, 3-collapsible
-  terrain_type = attr.ib(type=int, default=0)
```
[the remark this answers is not in the document]
Comment (dtch1997) [reply blind_walking/envs/locomotion_gym_config.py]: yeah this was a big oof from me. forgot to change it back. fixed.

File: blind_walking/envs/sensors/environment_sensors.py (hunk 2 above)
Comment (dtch1997) [reply blind_walking/envs/sensors/environment_sensors.py]: sure, implemented

File: blind_walking/net/feature_encoder.py (hunk 3 above)
Comment (dtch1997) [reply blind_walking/net/feature_encoder.py]: sure, implemented

File: blind_walking/net/feature_encoder.py (hunk 4 above)
Comment (dtch1997) [reply blind_walking/net/feature_encoder.py]: done

**Review by jennyzzt — APPROVED** [review]

**Pull request merged**